### PR TITLE
VIH-8898 add new scheduler job for hearing reminder in next 48 hrs

### PR DESCRIPTION
**Pull request by khizarkhanu-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-7318
https://tools.hmcts.net/jira/browse/VIH-8898

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by oliver-scott — CHANGES_REQUESTED** [review]

File: SchedulerJobs/SchedulerJobs.UnitTests/Services/HearingNotificationServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+namespace SchedulerJobs.UnitTests.Services
+{
+    public class HearingNotificationServiceTests
+    {
+        private Mock<IBookingsApiClient> _bookingApiClient;
+        private Mock<INotificationApiClient> _notificationApiClient;
+        private Mock<IFeatureToggles> _featureToggles;
+        private IHearingNotificationService _hearingNotificationService;
+        private Mock<ILogger<HearingNotificationService>> _logger;
+        private List<HearingDetailsResponse> nohearings;
```
[the remark this answers is not in the document]
Comment (oliver-scott) [reply SchedulerJobs/SchedulerJobs.UnitTests/Services/HearingNotificationServiceTests.cs:23]: So these would become

```
private List<HearingDetailsResponse> _noHearings;
private List<HearingDetailsResponse> _hearings;
private List<HearingDetailsResponse> _hearingsWithNoNotificationUserRoles;
private List<HearingDetailsResponse> _hearingsMultiple;
```

File: SchedulerJobs/SchedulerJobs.Services/HearingNotificationService.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            {
+                _logger.LogInformation("SendNotificationsAsync - No hearings to send notifications");
+                return;
+            }
+
+            await processHearings(hearings);
+
+        }
+
+        private async Task processHearings(List<HearingDetailsResponse> hearings)
```
[the remark this answers is not in the document]
Comment (khizarkhanu-hmcts) [reply SchedulerJobs/SchedulerJobs.Services/HearingNotificationService.cs]: Thanks Oliver, I have now fixed this.

File: SchedulerJobs/SchedulerJobs.UnitTests/Services/HearingNotificationServiceTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (khizarkhanu-hmcts) [reply SchedulerJobs/SchedulerJobs.UnitTests/Services/HearingNotificationServiceTests.cs]: Thanks, I have fixed this now.

File: SchedulerJobs/SchedulerJobs.UnitTests/Services/HearingNotificationServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+        private List<HearingDetailsResponse> hearings_1;
+        private List<HearingDetailsResponse> hearings_with_no_notification_userroles;
+        private List<HearingDetailsResponse> hearings_multiple;
+
+        [SetUp]
+        public void Setup()
+        {
+
```
[the remark this answers is not in the document]
Comment (khizarkhanu-hmcts) [reply SchedulerJobs/SchedulerJobs.UnitTests/Services/HearingNotificationServiceTests.cs]: Thanks, fixed it now.

**Review by oliver-scott — APPROVED** [review]

**Review by Anaia — APPROVED** [review]

**Pull request merged**